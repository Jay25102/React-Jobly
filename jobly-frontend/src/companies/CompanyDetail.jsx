import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import JoblyApi from "../../api";
import JobCardList from "../jobs/JobCardList";
import LoadingSpinner from "../LoadingSpinner";

/**
 * Displays info about company along with the jobs of that company.
 */
function CompanyDetail() {
    const { handle } = useParams();
    console.debug("CompanyDetail");

    const [company, setCompany] = useState(null);

    useEffect(function fetchCompanyOnMount() {
        console.debug("CompanyDetail useEffect fetchCompanyOnMount");
        async function fetchCompany() {
            setCompany(await JoblyApi.getCompany(handle));
        }
        fetchCompany();
    }, [handle]);

    if (!company) return <LoadingSpinner/>

    return (
        <div className="CompanyDetail col-md-8 offset-md-2">
            {company.name}
            {company.description}
            <JobCardList jobs={company.jobs}/>
        </div>
    )
}

export default CompanyDetail;