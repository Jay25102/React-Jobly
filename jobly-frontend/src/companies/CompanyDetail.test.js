import React from "react";
import { render } from "@testing-library/react";
import CompanyDetail from "./CompanyDetail";
import { UserProvider } from "../testUtils";

describe("JobCard", function() {
    test("it renders without crashing", function() {
        render(
            <UserProvider>
                <CompanyDetail/>                
            </UserProvider>
    );
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(
            <UserProvider>
                <CompanyDetail/>
            </UserProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
