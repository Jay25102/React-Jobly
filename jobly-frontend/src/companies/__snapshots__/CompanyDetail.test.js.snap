// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    loading...
  </div>
</DocumentFragment>
`;
